### ISSUE TITLE
Extensible predicates

### ISSUE THREAD

**Issue by hlship** [jkk/honeysql#62]
I'm doing some work with PostgreSQL and relative dates, so I'm creating some queries with clauses like:

```
[:< :expired_at (raw (format "now() - interval '%d seconds'" seconds))
```

I'd like to be able to replace those with something like:

```
[:< :expired_at [:before-now seconds]]
```

and have that expand to:

```
expired_at < now() - interval '5 seconds'
```

I guess this could just be a function that emits a raw, but doesn't seem full in the idiom of HoneySQL.

**Comment by seancorfield** [comment]
@hlship In MySQL, I'd expect `DATE_SUB( NOW(), INTERVAL ? SECONDS )` so I was curious about the quote in the PostgreSQL version: can you write `now() - interval '? seconds'` and still have a parameter there?

(I'm considering a new reader literal that is halfway between `raw` and regular constructed SQL so parameters would work -- but I'm not sure how parameter lifting would work if you have them inside SQL strings)

**Comment by seancorfield** [comment]
I think this might be partly addressed by #219 because once that's done you should be able to write:
```clojure
[:< :expired_at (raw ["now() - interval '" seconds " seconds'"])]
```

**Comment by seancorfield** [comment]
This has been implemented as part of #219 and the README now includes this particular example.

I'm not convinced that named abstractions/functions should be supported in the DSL (yet) so I'm going to close this out. Feel free to add comments to #219 for other use cases that are not yet supported by this.
